File: 0056-merge-intervals/0056-merge-intervals.java
class Solution {
    public int[][] merge(int[][] intervals) {
        Arrays.sort(intervals, (a,b)->a[0]-b[0]);
        int start = intervals[0][0];
        int end = intervals[0][1];
        List<int[]>mergedInterval = new ArrayList<>();
        for(int[] interval: intervals){
            if(interval[0] > end){
                mergedInterval.add(new int[]{start, end});
                start = interval[0];
                end = interval[1];
            }else{
                start = Math.min(start, interval[0]);
                end = Math.max(end, interval[1]);
            }
        }
        mergedInterval.add(new int[]{start, end});
        return mergedInterval.toArray(new int[0][mergedInterval.size()]);
    }
}

File: 0056-merge-intervals/0056-merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        sort(intervals.begin(),intervals.end());
        int start=intervals[0][0];
        int end=intervals[0][1];
        vector<vector<int>>ans;
        for(auto it:intervals){
            if(it[0]>end){
                vector<int>t(2);
                t[0]=start;
                t[1]=end;
                ans.push_back(t);
                start=it[0];
                end=it[1];
            }else{
                start=min(start,it[0]);
                end=max(end,it[1]);
            }
        }
        vector<int>t(2);
        t[0]=start;
        t[1]=end;
        ans.push_back(t);
        return ans;
    }
};